src/Loader/ServerVarResourceBaseLocator.php

```diff
@@ -5,27 +5,44 @@
 namespace Atoolo\Resource\Loader;
 
 use Atoolo\Resource\ResourceBaseLocator;
+use RuntimeException;
 
 class ServerVarResourceBaseLocator implements ResourceBaseLocator
 {
+    private ?string $resourceBase = null;
+
     public function __construct(
-        private readonly string $variableName,
-        private readonly ?string $subDirectory = null
+        private readonly string $variableName
     ) {
     }
 
     public function locate(): string
     {
+        if ($this->resourceBase !== null) {
+            return $this->resourceBase;
+        }
+
         if (empty($_SERVER[$this->variableName])) {
             throw new \RuntimeException(
                 'missing server variable ' . $this->variableName
             );
         }
 
-        if ($this->subDirectory === null) {
+        $resourceLayoutResourceBase = $_SERVER[$this->variableName] .
+            '/objects';
+        if (is_dir($resourceLayoutResourceBase)) {
+            $this->resourceBase = $resourceLayoutResourceBase;
+            return $resourceLayoutResourceBase;
+        }
+
+        if (is_dir($_SERVER[$this->variableName])) {
+            $this->resourceBase = $_SERVER[$this->variableName];
             return $_SERVER[$this->variableName];
         }
 
-        return $_SERVER[$this->variableName] . '/' . $this->subDirectory;
+        throw new RuntimeException(
+            "Resource root directory not found: " .
+            $_SERVER[$this->variableName]
+        );
```

Comment:
I think we can make this a bit easier to understand:

```suggestion

        $base = $_SERVER[$this->variableName];

        $baseObjects = $base . DIRECTORY_SEPARATOR . 'objects';
        if (is_dir($baseObjects)) {
            return $this->resourceBase = $baseObjects;
        }

        if (is_dir($base)) {
            return $this->resourceBase = $base;
        }

        throw new RuntimeException(
            'Resource root directory not found: ' . $base
        );

```